File: db/person_link.php
<?php

require_once("entity.php");

/**
 * Class for accessing, manipulating, creating and deleting records in person_link.
*/

class Person_Link extends Entity
{

  public function __construct($select = array())
  {
    $this->table = "person_link";
    $this->order = "";
    $this->domain = "person";
    $this->field['person_link_id']['type'] = 'SERIAL';
    $this->field['person_link_id']['not_null'] = true;
    $this->field['person_id']['type'] = 'INTEGER';
    $this->field['person_id']['not_null'] = true;
    $this->field['link_type_id']['type'] = 'INTEGER';
    $this->field['link_type_id']['not_null'] = true;
    $this->field['url']['type'] = 'VARCHAR';
    $this->field['url']['length'] = 1024;
    $this->field['url']['not_null'] = true;
    $this->field['title']['type'] = 'VARCHAR';
    $this->field['title']['length'] = 128;
    $this->field['description']['type'] = 'VARCHAR';
    $this->field['description']['length'] = 128;
    $this->field['rank']['type'] = 'INTEGER';
    $this->field['person_link_id']['primary_key'] = true;
    parent::__construct($select);
  }

}

?>

File: pentabarf/lib/momomoto/tables/person_link.rb
module Momomoto
  class Person_link < Base
    def initialize
      super
      @domain = 'person'
      @fields = {
        :person_link_id => Datatype::Integer.new( {:not_null=>true, :default=>true, :primary_key=>true, :serial=>true} ),
        :person_id => Datatype::Integer.new( {:not_null=>true} ),
        :link_type_id => Datatype::Integer.new( {:not_null=>true} ),
        :url => Datatype::Varchar.new( {:not_null=>true, :length=>1024} ),
        :title => Datatype::Varchar.new( {:length=>128} ),
        :description => Datatype::Varchar.new( {:length=>128} ),
        :rank => Datatype::Integer.new( {} )
      }
    end
  end
end
